pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1009,20 +1009,27 @@ type SemaphoreHolding struct {
 	Holders []string `json:"holders,omitempty" protobuf:"bytes,2,opt,name=holders"`
 }
 
-// MutexHolding holds the resource that holds the mutex lock.
+// MutexHolding describes the mutex and the object which is holding it.
 type MutexHolding struct {
-	// Name of the mutex
+	// Reference for the mutex
+	// e.g: ${namespace}/mutex/${mutexName}
 	Mutex string `json:"mutex,omitempty" protobuf:"bytes,1,opt,name=mutex"`
-	// HolderName stores the current mutex holder name.
-	HolderName string `json:"holderName,omitempty" protobuf:"bytes,2,opt,name=holderName"`
```

Comment:
is this existing? If so, then I think we cannot change the name unfortunately. We can enhance the comments.

Reply:
No, this is a new one.